api/v1/storageconsumer_types.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2020 Red Hat OpenShift Container Storage.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// StorageConsumerSpec defines the desired state of StorageConsumer
+type StorageConsumerSpec struct {
+	// Capacity is the total quota size allocated to a consumer.
+	// +optional
+	Capacity string `json:"capacity,omitempty"`
+}
+
+// ConfigSpec hold details of created objects required for external storage
+type ConfigSpec struct {
+	// RbdPoolName holds the name of created rbd pool.
+	RbdPoolName string `json:"rbdPoolName,omitempty"`
+	// CephUser holds the name of created ceph user.
+	CephUser string `json:"cephUser,omitempty"`
+}
+
+// StorageConsumerStatus defines the observed state of StorageConsumer
+type StorageConsumerStatus struct {
+	// State describes the state of StorageConsumer
+	State   string `json:"state,omitempty"`
```

Comment:
`State` should be an enum.  Planning to do that in the same PR?

Reply:
I would suggest the same PR as we could get complete picture of the API.

Reply:
Done